### ISSUE TITLE
Might want to consider black instead of yapf for auto-formatting

### ISSUE THREAD

**Issue by mulkieran** [stratis-storage/project#35]
yapf is not idempotent, but black is. It may have some other benefits. Not urgent, though.

**Comment by mulkieran** [comment]
Looks like it makes sense to get going with this, because of #58.

**Comment by mulkieran** [comment]
Partially blocked by #62.

**Comment by mulkieran** [comment]
@bgurney-rh , @jbaublitz, @waltdisgrace: At this point, there is a PR for this issue for every relevant repo in the stratis-storage project. Things I've noticed:

* black reformats on the order of 10% of every repo (except stratisd, where only the tests are reformatted)
* black reformats a bit more of stratis-cli, because stratis-cli has some very large dicts, which get entirely rewritten.
* black surrounds all strings with double quotes, so a fair amount of the formatting changes can be attributed to that decision.
* After black had run, I had to add about seven pylint annotation for the bad-continuation lint. However, indenting is not what pylint is best at and there are about 15 outstanding bugs about bad-continuations attached to the pylint repo, so this doesn't seem significant to me
* black moved one coverage pragma to the line after the line it was supposed to cover. However, it  can be forgiven, due to the "\" between the pragma and the line it was supposed to cover. This was easy to fix by hand, and black reformats the newly generated code just fine.

In summary, there seems to be little to worry about as a consequence of this switch to black, other than the fracture in the commit history. But if we're going to incur that at all, might as well do it as soon as possible.

Let me know if you have any reservations, otherwise I'll finalize the PRs and put them up for final review on Thursday.

**Comment by jbaublitz** [comment]
This seems completely reasonable to me. I'm biased, but I'd love to be able to put a guard for CI in my git hooks and if it's more reproducible locally, I'm all for it despite the git history fracture. The only point of concern I'd have is just the pylint aspect but based on what you're saying, even that seems really minor. Unless there's anything else, I'm for it.

**Comment by bgurney-rh** [comment]
In  tests/client-dbus/tests/udev/_crc32.py, it flattened the _CASTAGNOLI_TABLE into one-entry-per-line, going from 49 lines to 263 lines.

**Comment by mulkieran** [comment]
> In tests/client-dbus/tests/udev/_crc32.py (in the stratisd PR), it flattened the _CASTAGNOLI_TABLE into one-entry-per-line, going from 49 lines to 263 lines.

I think I'm cool w/ that...it's not something I wanted to read to begin with. And it is consistent w/ their stated rule.
